### ISSUE TITLE
bug: @ionic/vue-router does not detect a replaced route outside of guard contexts

### ISSUE THREAD

**Issue by kas84** [ionic-team/ionic-framework#22412]
# Bug Report

**Ionic version:**

[ ] **4.x**
[x] **5.x**

**Current behavior:**

Replacing routes so  we only have one current route and nowhere to go back but swiping back still works.

**Expected behavior:**

Souldn't be able to go back since window.history.back == null
Something like the following coded that worked on v1 would work just fine.
```
    $ionicViewService.nextViewOptions({
      disableBack: true
    });
```
**Steps to reproduce:**

Land on a page, on mounted replace the route with another one. On this last one, swipe back.

**Other information:**

**Ionic info:**

```

Ionic:

   Ionic CLI       : 6.12.0 (/Users/***/.nvm/versions/node/v14.7.0/lib/node_modules/@ionic/cli)
   Ionic Framework : @ionic/vue 5.4.0

Capacitor:

   Capacitor CLI   : 2.4.2
   @capacitor/core : 2.4.2

Utility:

   cordova-res : 0.15.1
   native-run  : not installed

System:

   NodeJS : v14.7.0 (/Users/***/.nvm/versions/node/v14.7.0/bin/node)
   npm    : 6.14.8
   OS     : macOS Catalina

```

**Comment by kas84** [comment]
Here's the repo with the bug reproduced.

https://github.com/kas84/ionic-vue-router-bug

Here's a gif that illustrates the bug:
![ionic-vue-router-bug](https://user-images.githubusercontent.com/5584/97913635-e0bbf100-1d4e-11eb-81ed-2d0e551d0b12.gif)

**Comment by liamdebeasi** [comment]
Thanks for the issue. Can you try the following dev build and let me know if it resolves the issue?

```
npm install @ionic/vue@5.5.0-dev.202011041535.a9b2260 @ionic/vue-router@5.5.0-dev.202011041535.a9b2260
```

**Comment by kas84** [comment]
Yes. Works fine. Thanks!!!

**Comment by liamdebeasi** [comment]
Great! I am going to keep this issue open until https://github.com/ionic-team/ionic-framework/pull/22429 gets merged.

**Comment by liamdebeasi** [comment]
Thanks for the issue. This has been resolved via https://github.com/ionic-team/ionic-framework/pull/22429, and a fix will be available in an upcoming release of Ionic Framework.

**Comment by kas84** [comment]
Hi! Is this merged in to a new relase of ionic-vue? Also, let's say you have a login page that is divided in two pages... first email, then password, son on first page, you push, and on the second page you replace the route if the login was successfull... is this supposed to work in a way that swipe is not enabled? 

On ionic v1 you had 

 ```
$ionicViewService.nextViewOptions({
      disableAnimate: true,
      disableBack: true
});
```

How can you achieve the same thing, here?

**Comment by liamdebeasi** [comment]
@kas84 This fix was released as part of Ionic Vue 5.4.2

Regarding your question, I recommend posting it on our forums: http://forum.ionicframework.com/
